#157965217 Create API Endpoints

asamisellah opened the pull request:
#### What does this PR do?
Create CRUD API Endpoints 

#### Description of tasks to be completed?
METHOD | ENDPOINT | FUNCTIONALITY 
------------ | -------------|--------------
GET | /api/v1/users/requests/ | Fetch all the requests resources
GET | /api/v1/users/requests/requestId | Fetch a single request resource
PUT | /api/v1/users/requests/requestId | Modify a request
POST | /api/v1/users/requests | Create a new request
DELETE | /api/v1/users/requests/requestId | Delete a specific request
GET | /api/v1/users/signup/ | Handle user registration
POST | /api/v1/users/signin/ | Handle user login

#### How should this be manually tested?
1. clone the [repo](https://github.com/asamisellah/Maintenance-Tracker)
2. open the folder in your text editor or IDE
3. `git checkout -b new-branch`
4. `git fetch ch-env-setup-157930685`

*Setup*
Make sure virtual environment is installed. Open terminal and run the following commands
1. `cd maintenance-tracker`
2.  `virtualenv venv` to create virtual environment
3.  `source venv/bin/activate` to run environment
4. `pip install -r requirements.txt`

*Testing*
run ` pytest`

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#157965217 ](https://www.pivotaltracker.com/story/show/157965217/comments/190109380)
[#157995248](https://www.pivotaltracker.com/story/show/157995248/comments/190191728)

### Screenshot
![signin_user](https://user-images.githubusercontent.com/13402544/40820555-e1fdb14c-6569-11e8-9342-1f45e199e020.png)
Create User

![create_user](https://user-images.githubusercontent.com/13402544/40820535-ce83ae6e-6569-11e8-94e2-dd09dc0d156f.png)
Sign In user

![post_request](https://user-images.githubusercontent.com/13402544/40820591-171b602c-656a-11e8-8342-b7c7fa43673d.png)
Create Request

![get_requests](https://user-images.githubusercontent.com/13402544/40820621-403c2e78-656a-11e8-897d-990fdde86b44.png)
Get  #Requests

### Questions
Have I satisfied feedback
Are the endpoints functioning?

asamisellah committed:
[Feature #157965217]Structure endpoint routes and methods

asamisellah committed:
[Feature #157965217]Create get all users endpoint

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,58 @@
+
+
+# Get all users
+@app.route('/user')
```
From the rubric posted you are not supposed to have a route for getting all users
https://docs.google.com/document/d/1rZPo0ONvZco_ILlNs62UZNwrZCtNsyqU9IEvjKNlSp8/

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,58 @@
 
 app = Flask(__name__)
+
+user = [
```
Don't use hard coded data

asamisellah committed:
[Feature #157965217] Create API Endpoints
The following are the endpoints created:
- Get users/requests
- Get users/requests/requestId
- Put users/requests/requestId
- Post users/requests

asamisellah committed:
[Feature #157965217] Added endpoints
- GET /users
- POST /users
- DELETE /users/requests/<requestId> endpoints

asamisellah committed:
[Chore #157995248] Pass test for endpoint /users/requests

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,100 @@
 app = Flask(__name__)
+
+
+requests = {}
```
requests should be a list.

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,100 @@
+
+requests = {}
+
+users = {}
```
users too

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,100 @@
+        "password": request.json.get("password"),
+        "cconfirm_password": request.json.get('cconfirm_password'),
+    }
+    users.update(user)
```
users.append

zmwas commented on app/__init__.py:
```diff
@@ -1,3 +1,100 @@
+@app.route('/users/requests', methods=['POST'])
+def create_request():
+    # "id" = request.json.get("id")
+    request_data = {request.json.get("id"): {
```
The json is structured wrong, it should be `{"id":len(requests)+1,
        "title": request.json.get("title"),
        "type": request.json.get('type'),
        "description": request.json.get("description"),
        "category": request.json.get('category'),
        "area": request.json.get('area')
    }`

zmwas commented on test_requests.py:
```diff
@@ -1,41 +1,45 @@
-    def test_update_request(self):
-        res = self.client.put('/users/requests/1', data=self.request)
-        self.assertEqual(res.status_code, 200)
+    # def test_get_request(self):
```
Uncomment out the code

zmwas commented:
Also update your PR to have screenshots of the POSTMAN requests for the endpoints.
